query/src/main/scala/filodb/query/exec/rangefn/MiscellaneousFunction.scala

```diff
@@ -0,0 +1,81 @@
+package filodb.query.exec.rangefn
+
+import java.util.regex.{Pattern, PatternSyntaxException}
+
+import monix.reactive.Observable
+
+import filodb.core.query.{CustomRangeVectorKey, IteratorBackedRangeVector, RangeVector, RangeVectorKey}
+import filodb.memory.format.ZeroCopyUTF8String
+
+trait MiscellaneousFunction {
+  def execute(source: Observable[RangeVector]): Observable[RangeVector]
+}
+
+case class LabelReplaceFunction(funcParams: Seq[Any])
+  extends MiscellaneousFunction {
+
+  val labelIdentifier: String = "[a-zA-Z_][a-zA-Z0-9_:\\-\\.]*"
+
+  require(funcParams.size == 4,
+    "Cannot use LabelReplace without function parameters: " +
+      "instant-vector, dst_label string, replacement string, src_label string, regex string")
+
+  val dstLabel: String = funcParams(0).asInstanceOf[String]
+  val replacementString: String = funcParams(1).asInstanceOf[String]
+  val srcLabel: String = funcParams(2).asInstanceOf[String]
+  val regexString: String = funcParams(3).asInstanceOf[String]
+
+  require(dstLabel.matches(labelIdentifier), "Invalid destination label name")
+
+  try {
+    Pattern.compile(regexString)
+  }
+  catch {
+    case ex: PatternSyntaxException => {
+      throw new PatternSyntaxException("Invalid Regular Expression for label_replace\n" +
+        ex.getDescription, regexString, ex.getIndex)
```

Comment:
Dont lose the original exception. It may be simplest to do
```
throw new IllegalArgumentException("Invalid Regular Expression for label_replace", ex)
```

Reply:
Done!